include/swift/Sema/ConstraintSystem.h

```diff
@@ -3438,6 +3438,10 @@ class ConstraintSystem {
     return findSelectedOverloadFor(calleeLoc);
   }
 
+  unsigned int getCurrentIndent() const {
```

Comment:
The reason I decided to move this logic to the ConstraintSystem was that I noticed a couple of places like [this one](https://github.com/apple/swift/blob/fc954d299b41873fda5f0e74e37b8cc058f39ecf/lib/Sema/ConstraintGraph.cpp#L1137-L1138) and [this one](https://github.com/apple/swift/blob/fc954d299b41873fda5f0e74e37b8cc058f39ecf/lib/Sema/ConstraintSystem.cpp#L3574) with similar code that was checking for `solverState` existence. Since I needed to fetch indentation in numerous other places, it felt like moving this logic to ConstraintSystem would reduce duplication.  

I am not sure how to employ `getDebugLogger()` approach here. There is still a need to get the indentation to pass it on [to print constraints](https://github.com/apple/swift/blob/fc954d299b41873fda5f0e74e37b8cc058f39ecf/lib/Sema/CSStep.h#L538-L540).

I can leave the `getCurrentIndent()` refactoring out of the scope of this pull request and revert to checking for `solverState` externally where appropriate.

Reply:
> please separate refactoring changes

I split this PR into three commits. If it is too large, I can break it down into several PRs.

Reply:
> it's effectively NFC refactoring and could be done separately

I reverted the `getCurrentIndent()` refactoring